doc/design_AsyncNDRegisterAcessor_and_NumericAddressedBackend.dox

```diff
@@ -0,0 +1,60 @@
+
+\subsection  async_NumericAddressedBackend_interface Interface for backend implementations
+
+The NumericAddressedInterruptDispatcher serves two purposes:
+\li It serves as a subscription manager for the AsyncNDRegisterAccessor
+\li It provides functions which act on all subsribers.
+
+When the map file is read in, a NumericAddressedInterruptDispatcher is created for each interrupt.
+
+The only function the backend implementor really has to deal with is NumericAddressedInterruptDispatcher::trigger(). All other functionaltiy for subscription, activation/deactivation and exception handling is already handled in the NumericAddressedBackend.
+
+There is one NumericAddressedInterruptDispatcher per interrupt controller number and interrupt number, stored in the variable NumericAddressedBackend::_interruptDispatchers. The interrupt handling thread in the backend implementation will just call
+
+<pre>
+  _interruptDispatcher.at({interruptControllerNumber, interruptNumber})->trigger();
+</pre>
+
+Notice that you should access the map with at() and not through the [ ] operator, so you don't accidentally add empty dispatchers to the map.
+The interrupt handler thread should have a "lazy start", which means that it should only be started once accessors to the particular interrupt have been requested. To achieve this, the backend should override NumericAddressedBackend::startInterruptHandlingThread(unsigned int interruptControllerNumber, unsigned int interruptNumber). This function is called each time an asynchronous accessor which is associated to that interrupt is created. The implementing code has to make sure that the thead is only started once. The thread in not terminated automatically. The backend has to take care to do this when the backend is closed and/or destructed.
+
+Internally an object called NumericAddressedAsyncVariable (or to be more precie a shared pointer to the typed implementation NumericAddressedAsyncVariableImpl) is stored for each required accessor type. This contains one synchronous read accessor together with all its subscribers. Like this the hardware access to the devide and the possibly expensive postRead() action (de-mupltiplexing and numeric conversion) are only done once, before the data is pushed into the data transport queues of the subscribers.
+There can be multiple NumericAddressedAsyncVariables for one register. For example one for raw access, one with data conversion and a thrid one for only the first 10 elements in a large 1D register. All this is happening automatically in the NumericAddressedBackend and the developer of the individual backend does not have to care about it.
+
+\subsection design_async_map_file Asynchronous registers in the map file
```

Comment:
Maybe move this subsection up, since you are referring to the interrupt controller and interrupt number concepts in the previous subsection already.